scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/SchedulerAppDriver.java

```diff
@@ -61,23 +74,36 @@ public static void main(final String[] args) {
     javalin.start(config.httpPort());
   }
 
-  /**
-   * collects configuration options from the environment
-   *
-   * any options not specified in the input stream fall back to the hard-coded defaults here
-   *
-   * @return a complete configuration object reflecting choices elected in the environment or the defaults
-   */
-  private static AppConfiguration loadConfiguration() {
-    return new AppConfiguration(
-        Integer.parseInt(getEnvOrFallback("SCHED_PORT", "27193")),
-        Boolean.parseBoolean(getEnvOrFallback("SCHED_LOGGING", "true")) ?
-            JavalinLoggingState.Enabled : JavalinLoggingState.Disabled,
-        URI.create(getEnvOrFallback("MERLIN_GRAPHQL_URL", "http://localhost:8080/v1/graphql")),
-        Path.of(getEnvOrFallback("MERLIN_LOCAL_STORE", "/usr/src/app/merlin_file_store")),
-        Path.of(getEnvOrFallback("SCHED_RULES_JAR", "/usr/src/app/merlin_file_store/sched_rules.jar")),
-        PlanOutputMode.valueOf((getEnvOrFallback("SCHED_OUTPUT_MODE", "CreateNewOutputPlan")))
-    );
+  private record Stores(SpecificationRepository specifications, ResultsCellRepository results) { }
+
+  private static Stores loadStores(final AppConfiguration config) {
+    final var store = config.store();
+    if (store instanceof final PostgresStore pgStore) {
+      final var pgDataSource = new PGDataSource();
+      pgDataSource.setServerName(pgStore.server());
+      pgDataSource.setPortNumber(pgStore.port());
+      pgDataSource.setDatabaseName(pgStore.database());
+      pgDataSource.setApplicationName("Scheduler Server");
+
+      final var hikariConfig = new HikariConfig();
+      hikariConfig.setUsername(pgStore.user());
+      hikariConfig.setPassword(pgStore.password());
+      hikariConfig.setDataSource(pgDataSource);
+
+      final var hikariDataSource = new HikariDataSource(hikariConfig);
+
+      return new Stores(
+          new PostgresSpecificationRepository(hikariDataSource),
+          new PostgresResultsCellRepository(hikariDataSource));
+    } else if (store instanceof InMemoryStore) {
+      final var inMemoryPlanRepository = new InMemorySpecificationRepository();
```

Comment:
Similar to Adrien's comment - this variable has the word plan in it, but should probably say specification instead.

Reply:
Just pushed this change, good catch!